From b75bee7c34e054448814d749d32ff5f053702a59 Mon Sep 17 00:00:00 2001
From: David Mercado <david.mercado@windriver.com>
Date: Wed, 20 Nov 2013 17:02:41 -0500
Subject: [PATCH 151/175] LSI ACP34XX: USB: Do not map secondary register region for PowerPC

git.yoctoproject.org/git/linux-yocto-3.10
commit f9fdbcefe0cf40de42d6a4697cfd6dbd2cc3e807 standard/axxia/base.

The ACP34xx device tree does not specify a secondary register region.
Do not map this region in for the ACP34xx to avoid an error during
driver initialization.

Signed-off-by: Hong H. Pham <hong.pham@windriver.com>
---
 drivers/usb/host/ehci-ci13612.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/host/ehci-ci13612.c b/drivers/usb/host/ehci-ci13612.c
index 8e4a1f3..b39dc0e 100644
--- a/drivers/usb/host/ehci-ci13612.c
+++ b/drivers/usb/host/ehci-ci13612.c
@@ -227,7 +227,9 @@ static int ci13612_ehci_probe(struct platform_device *pdev)
 {
 	struct device_node *np = pdev->dev.of_node;
 	struct usb_hcd *hcd;
+#ifdef CONFIG_ARM
 	void __iomem *gpreg_base;
+#endif
 	int irq;
 	int retval;
 	struct resource *res;
@@ -275,6 +277,7 @@ static int ci13612_ehci_probe(struct platform_device *pdev)
 		goto fail_put_hcd;
 	}
 
+#ifdef CONFIG_ARM
 	gpreg_base = of_iomap(np, 1);
 	if (!gpreg_base) {
 		dev_warn(&pdev->dev, "of_iomap error can't map region 1\n");
@@ -285,6 +288,7 @@ static int ci13612_ehci_probe(struct platform_device *pdev)
 		writel(0x0, gpreg_base + 0x8);
 		iounmap(gpreg_base);
 	}
+#endif
 
 	retval = usb_add_hcd(hcd, irq, 0);
 	if (retval == 0) {
-- 
1.7.1

